docs/examples/DataSet/DataSet-class-walkthrough.ipynb

```diff
@@ -0,0 +1,1327 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# DataSet class walkthrough\n",
+    "\n",
+    "In this notebook we are going to go through the mighty `DataSet` class, and get an overview of the most of it's methods and properties. In this case, the more you know, the more you can achieve."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Note on DataSet object creation\n",
+    "\n",
+    "Users are expected obtain `DataSet` objects in the following ways:\n",
```

Comment:
Are these expectations or the only correct ways? I mean, can the Dataset objects be obtained via different methods that are not expected but still valid?

Reply:
so our users shall obtain DataSet objects in these ways. qcodes devs may use other means within qcodes internals, but users should not. Is this clear? also, shall i reformulate this section in some way to make this more clear?

Reply:
Ok I get it now. Then the word "expectation" makes sense indeed. I am satisfied, so it is your decision to revise :)